packages/front-end/components/Experiment/AnalysisSettingsBar.tsx

```diff
@@ -315,11 +331,17 @@ export default function AnalysisSettingsBar({
                         }),
                       })
                         .then(() => {
+                          trackSnapshot("create", {id: snapshot?.id || "", ...snapshotProps});
```

Comment:
`snapshot` here refers to the old snapshot on the page, not the new one you just created.  Instead, you need to get it from the API response:

```ts
.then((res) => {
  const id = res.snapshot?.id || "";
```